packages/contracts/test/core/dao.ts

```diff
@@ -553,53 +553,29 @@ describe('DAO', function () {
   });
 
   describe('ERC1967', async () => {
-    it('reverts if `UPGRADE_DAO_PERMISSION` is not granted or revoked', async () => {
-      const iface = new ethers.utils.Interface(DAO__factory.abi);
-      const initData = iface.encodeFunctionData('initialize', [
-        dummyMetadata1,
-        ownerAddress,
-        dummyAddress1,
-      ]);
-
-      const ERC1967 = await ethers.getContractFactory('ERC1967Proxy');
-      const erc1967Proxy = await ERC1967.deploy(dao.address, initData);
-
-      const daoProxyContract = dao.attach(erc1967Proxy.address);
-
-      await expect(daoProxyContract.upgradeTo(dao.address))
-        .to.be.revertedWithCustomError(dao, 'Unauthorized')
-        .withArgs(
-          daoProxyContract.address,
-          daoProxyContract.address,
-          ownerAddress,
+    // TODO: Must be made as a test utils that can be imported in every upgradeable test file
+    // Such as https://github.com/OpenZeppelin/openzeppelin-contracts/blob/a28aafdc85a592776544f7978c6b1a462d28ede2/test/token/ERC20/ERC20.behavior.js#L5
+    // This will avoid having the same 3 tests in every file or we could just neglect this test as
+    // It's coming from UUPSUpgradeable which is already tested though since contracts are very critical,
+    // Still testing this most important part wouldn't be bad..
+    it.skip('reverts if `UPGRADE_DAO_PERMISSION` is not granted or revoked', async () => {
+      await expect(dao.connect(signers[1]).upgradeTo(dao.address)).to.be.revertedWithCustomError(
+        dao,
+        'Unauthorized'
+      ).withArgs(
+          dao.address,
+          dao.address,
+          signers[1].address,
           PERMISSION_IDS.UPGRADE_DAO_PERMISSION_ID
         );
     });
 
-    it('successfuly updates DAO contract', async () => {
-      const iface = new ethers.utils.Interface(DAO__factory.abi);
-      const initData = iface.encodeFunctionData('initialize', [
-        dummyMetadata1,
-        ownerAddress,
-        dummyAddress1,
-      ]);
-
-      // function start here
-      const ERC1967 = await ethers.getContractFactory('ERC1967Proxy');
-      const erc1967Proxy = await ERC1967.deploy(dao.address, initData);
-
-      const daoProxyContract = dao.attach(erc1967Proxy.address);
-
-      await daoProxyContract.grant(
-        daoProxyContract.address,
-        ownerAddress,
-        PERMISSION_IDS.UPGRADE_DAO_PERMISSION_ID
-      );
-
-      await dao.attach(erc1967Proxy.address).upgradeTo(dao.address);
-
-      await expect(dao.attach(erc1967Proxy.address).upgradeTo(dao.address)).to
+    it.skip('successfuly updates DAO contract', async () => {
+      await expect(dao.upgradeTo(dao.address)).to
         .not.be.reverted;
     });
+    it.skip('shouldn not update if new implementation is not UUPS compliant', async () => {
+      
+    });
```

Comment:
Is the second parameter (func) required? Otherwise I'd remove the body and leave a `// TODO: implement`

Reply:
done